test/test-mongocrypt-crypto-hooks.c

```diff
@@ -730,6 +782,118 @@ _test_crypto_hook_sign_rsaes_pkcs1_v1_5 (_mongocrypt_tester_t *tester)
    bson_string_free (call_history, true);
 }
 
+#ifdef MONGOCRYPT_ENABLE_CRYPTO_LIBCRYPTO
```

Comment:
Consider also testing by parameterizing `_test_crypto_aes_256_ctr` to run with and without an ECB hook. That may help to verify against more data.

Reply:
You were talking about `_test_native_crypto_aes_256_ctr`, weren't you?

Reply:
Oops, yes.

Reply:
I considered it, but I was trying to avoid exposing unnecessary symbols. I already had to expose `_native_crypto_aes_256_ecb_encrypt`, which isn't serving any purpose but testing. I guess I could have copied `_encrypt_with_cipher` to the test file, but that looks like too much copy/paste to me. In order to run CTR test vectors on `_crypto_aes_256_ctr_encrypt_decrypt_via_ecb` I will have to expose it, e.g.: make it non-static and symbol will become visible. That was my motivation for using `_mongocrypt_fle2_do_encryption` in order to get the coverage - it would use `_crypto_aes_256_ctr_encrypt_decrypt_via_ecb` and I won't need to make it public.

Let me know what you think. If making `_crypto_aes_256_ctr_encrypt_decrypt_via_ecb` public is no big deal, I shall do it

Reply:
Good point. I had not considered that `_crypto_aes_256_ctr_encrypt_decrypt_via_ecb` would need to be made public. Resolving.